shark/examples/eager_mode.py

```diff
@@ -0,0 +1,123 @@
+import torch
+from torch.nn import Parameter
+from torch.utils.cpp_extension import load_inline, include_paths
+from torch_mlir.eager_mode import torch_mlir_tensor
+from torch_mlir.eager_mode.torch_mlir_tensor import TorchMLIRTensor
+
+from shark.iree_eager_backend import EagerModeIREELinalgOnTensorsBackend
+from shark.shark_runner import SharkMode
+
+N = 10
+
+
+def __new__(cls, data=None, requires_grad=True):
+    if data is None:
+        data = torch.empty(0)
+    if isinstance(data, TorchMLIRTensor):
+        return data
+    else:
+        return torch.Tensor._make_subclass(cls, data, requires_grad)
+
+
+Parameter.__new__ = __new__
+
+
+def test_cpu():
+    torch_mlir_tensor.backend = EagerModeIREELinalgOnTensorsBackend("cpu")
+
+    t = torch.ones((N, N), device="cpu")
+    u = 2 * torch.ones((N, N), device="cpu")
+
+    tt = TorchMLIRTensor(t)
+    print(tt)
+    uu = TorchMLIRTensor(u)
+    print(uu)
+
+    for i in range(100):
+        yy = tt + uu
+        print(yy.elem.to_host())
+        yy = tt * uu
+        print(yy.elem.to_host())
+
+
+def test_gpu():
+    source = """
+    #include <iostream>
+    #include "cuda.h"
+    #include "cuda_runtime_api.h"
+
+    using namespace std;
+
+    void print_free_mem() {
+        int num_gpus;
+        size_t free, total;
+        cudaSetDevice(0);
+        int id;
+        cudaGetDevice(&id);
+        cudaMemGetInfo(&free, &total);
+        cout << "GPU " << id << " memory: used=" << (total-free)/(1<<20) << endl;
+    }
+    """
+    gpu_stats = load_inline(
+        name="inline_extension",
+        cpp_sources=[source],
+        extra_include_paths=include_paths(cuda=True),
+        functions=["print_free_mem"],
+    )
+    torch_mlir_tensor.backend = EagerModeIREELinalgOnTensorsBackend("cuda")
+
+    t = torch.ones((N, N), device="cpu")
+    u = 2 * torch.ones((N, N), device="cpu")
+
+    tt = TorchMLIRTensor(t)
+    print(tt)
+    uu = TorchMLIRTensor(u)
+    print(uu)
+
+    for i in range(100):
+        yy = tt + uu
+        print(yy.elem.to_host())
+        yy = tt * uu
+        print(yy.elem.to_host())
+        gpu_stats.print_free_mem()
+
+
+def test_python_mode_ref_backend():
+    # hide this wherever you want?
+    _ = SharkMode()
+
+    t = torch.ones((N, N), device="cpu")
+    u = torch.ones((N, N), device="cpu")
+
+    print(t)
+    print(u)
+
+    for i in range(100):
+        yy = t + u
+        print(yy.elem)
+        yy = t * u
+        print(yy.elem)
+
+
+def test_python_mode_iree():
+    for device in ["cpu", "cuda"]:
+        _ = SharkMode(device)
+
+        t = torch.ones((N, N), device="cpu")
+        u = torch.ones((N, N), device="cpu")
+
+        print(t)
+        print(u)
+
+        for i in range(5):
+            yy = t + u
+            print(yy.elem.to_host())
+            yy = t * u
+            print(yy.elem.to_host())
+
+
+if __name__ == "__main__":
+    # test_cpu()
+    # test_gpu()
+    test_python_mode_ref_backend()
+    # test_python_mode_iree()
```

Comment:
A lot of these functions are never called?

Reply:
they're all for exercising the iree backend, which is blocked

Reply:
I see, is the intention to land this once they are unblocked?

Reply:
Latest shark build should have a fix

Reply:
actually, if IREE is unblocked now should this be uncommented?

Reply:
nevermind was looking at old stuff